File: 007_Reverse_Integer/solution.go
const (
    MAX int = 1<<31 - 1
)

func reverse(x int) int {
    sign := 1
    tmp := x
    if x < 0 {
        sign = -1
        tmp = -x
    }
    list := []int{}
    for tmp > 9 {
        list = append(list, tmp % 10)
        tmp = tmp / 10
    }
    list = append(list, tmp)
    ret := 0
    for i := 0; i < len(list); i ++ {
        ret = 10 * ret + list[i]
        if ret > MAX {
            return 0
        }
    }
    return ret * sign
}

File: 007_Reverse_Integer/solution.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        sign = 1 if x > 0 else -1
        x = abs(x)
        res = 0
        while x > 0:
            res = res * 10 + x % 10
            if res > math.pow(2, 31):
                return 0
            x = x / 10
        return res * sign
